src/common/types/uuid.cpp

```diff
@@ -0,0 +1,69 @@
+#include "duckdb/common/types/uuid.hpp"
+
+#include <sstream>
+#include <iomanip>
+#include <inttypes.h>
+
+namespace duckdb {
+
+bool UUID::FromString(string str, hugeint_t &result) {
+	auto hex2char = [](char ch) -> unsigned char {
+		if (ch >= '0' && ch <= '9') {
+			return ch - '0';
+		}
+		if (ch >= 'a' && ch <= 'f') {
+			return 10 + ch - 'a';
+		}
+		if (ch >= 'A' && ch <= 'F') {
+			return 10 + ch - 'A';
+		}
+		return 0;
+	};
+	auto is_hex = [](char ch) -> bool {
+		return (ch >= '0' && ch <= '9') || (ch >= 'a' && ch <= 'f') || (ch >= 'A' && ch <= 'F');
+	};
+
+	if (str.empty()) {
+		return false;
+	}
+	int has_braces = 0;
+	if (str.front() == '{') {
+		has_braces = 1;
+	}
+	if (has_braces && str.back() != '}') {
+		return false;
+	}
+
+	result.lower = 0;
+	result.upper = 0;
+	size_t count = 0;
+	for (size_t i = has_braces; i < str.size() - has_braces; ++i) {
+		if (str[i] == '-') {
+			continue;
+		}
+		if (count >= 32 || !is_hex(str[i])) {
+			return false;
+		}
+		if (count >= 16) {
+			result.lower = (result.lower << 4) | hex2char(str[i]);
+		} else {
+			result.upper = (result.upper << 4) | hex2char(str[i]);
+		}
+		count++;
+	}
+	return count == 32;
+}
+
+void UUID::ToString(hugeint_t input, char *buff) {
+	std::snprintf(buff, 36, "%08" PRIx64 "-%04" PRIx64 "-%04" PRIx64 "-%04" PRIx64 "-%011" PRIx64,
```

Comment:
@Mytherin Please help review this function, not sure if there is a better way to do this.

For the formatting, I can't use `%ldx` or `%lux` as the build will fail in some linux platform.

Reply:
Perhaps do the conversion yourself instead of using snprintf, e.g.:

```cpp
void ByteToHex(char byte_val, char *buf, idx_t &pos) {
	static char const HEX_DIGITS[] = "0123456789abcdef";

	buf[pos++] = HEX_DIGITS[(byte_val >> 4) & 0xf];
	buf[pos++] = HEX_DIGITS[byte_val & 0xf];
}

void UUID::ToString(hugeint_t input, char *buf) {
    idx_t pos = 0;
    ByteToHex(input.upper >> 32 & 0xFFFFFFFF , buf, pos);
    ...
    buf[pos++] = '-';
    ...
}
```

Will be faster as well :)